26578 - feat: disable business summary for certain corp types

hfekete opened the pull request:
*Issue:*https://github.com/bcgov/entity/issues/

*Description of changes:*
- disable business summary for certain corp types and their continuation in countrerparts (BC, CC, ULC)
- update tooltip to match.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

hfekete commented:
Flag is updated for dev so it can be tested in dev. When merged, test flag should be updated too, so that QAs can test.

hfekete commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://business-dashboard-dev--pr-176-ar99hav8.web.app

kialj876 commented:
Looks good to me. Just need to verify the cypress test failure

JazzarKarim reviewed (approved):
LGTM. Can you update the package version number Hrvoje please?

severinbeauvais commented:
Can you please provide a couple of screenshots and/or some test URLs? Thanks.

hfekete commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://business-dashboard-dev--pr-176-ar99hav8.web.app

hfekete commented:
> LGTM. Can you update the package version number Hrvoje please?

thanks for reminding me this

hfekete commented:
> Can you please provide a couple of screenshots and/or some test URLs? Thanks.

added, except CCC continuation in

severinbeauvais commented:
Here's the most important test business, a BEN, who _should_ have the Business Summary enabled 😁 

https://business-dashboard-dev--pr-176-ar99hav8.web.app/BC1230018

severinbeauvais commented:
> And a continued-in BEN:
> 
> https://business-dashboard-dev--pr-176-ar99hav8.web.app/C9900819

Actually, for this one, the Business Summary should be hidden (since most actions are hidden by another FF).

severinbeauvais commented:
Another example. Why is the Business Summary disabled for this continued-in BEN?

https://business-dashboard-dev--pr-176-ar99hav8.web.app/C9900833

hfekete commented:
> > And a continued-in BEN:
> > https://business-dashboard-dev--pr-176-ar99hav8.web.app/C9900819
> 
> Actually, for this one, the Business Summary should be hidden (since most actions are hidden by another FF).

if we want to hide some, and disable others, we will need two flags, or hardcode some information

hfekete commented:
> ~Another example. Why is the Business Summary disabled for this continued-in BEN?~
> 
> https://business-dashboard-dev--pr-176-ar99hav8.web.app/C9900833
> 
> _Update: this is disabled because CBEN is not configured as one of the support entity types (via FF). Should it be? In fact what FF values do you want for Dev/Test/Prod? What about Prod on April 1?_

just wanted to say the same, I am not sure which ones should be available, what was done in test is I asked Kial to remove the ones to be removed and specified in ticket for this.

severinbeauvais commented:
> > > And a continued-in BEN:
> > > https://business-dashboard-dev--pr-176-ar99hav8.web.app/C9900819
> > 
> > 
> > Actually, for this one, the Business Summary should be hidden (since most actions are hidden by another FF).
> 
> if we want to hide some, and disable others, we will need two flags, or hardcode some information

No, I think it's because you removed the `!isDisableNonBenCorps()` check.

severinbeauvais commented:
> > > And a continued-in BEN:
> > > https://business-dashboard-dev--pr-176-ar99hav8.web.app/C9900819
> > 
> > 
> > Actually, for this one, the Business Summary should be hidden (since most actions are hidden by another FF).
> 
> if we want to hide some, and disable others, we will need two flags, or hardcode some information

No, I think it's because you removed the `!isDisableNonBenCorps()` check.

Well, you moved that check to make the button disabled. What is your requirement? I'm thinking this action should be hidden, same as before.

hfekete commented:
> > > > And a continued-in BEN:
> > > > https://business-dashboard-dev--pr-176-ar99hav8.web.app/C9900819
> > > 
> > > 
> > > Actually, for this one, the Business Summary should be hidden (since most actions are hidden by another FF).
> > 
> > 
> > if we want to hide some, and disable others, we will need two flags, or hardcode some information
> 
> No, I think it's because you removed the `!isDisableNonBenCorps()` check.

I moved that one to the disabled, together with flag check. Should I keep that in hidden and the other ones in disabled ?

severinbeauvais commented:
> just wanted to say the same, I am not sure which ones should be available, what was done in DEV FF is I asked Kial to remove the ones to be removed and specified in ticket for this.

I need to know, as I am updating the FFs on April 1 to enable the Early Adopters release. Please find out and let me know here or in ticket bcgov/entity#26268. Thanks.

severinbeauvais commented:
> I moved that one to the disabled, together with flag check. Should I keep that in hidden and the other ones in disabled ?

If we want to keep the previous functionality then you need to move it back, but I don't know if your requirements supersede that. Please find out.

hfekete commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://business-dashboard-dev--pr-176-ar99hav8.web.app

hfekete commented:
> > just wanted to say the same, I am not sure which ones should be available, what was done in DEV FF is I asked Kial to remove the ones to be removed and specified in ticket for this.
> 
> I need to know, as I am updating the FFs on April 1 to enable the Early Adopters release. Please find out and let me know here or in ticket [bcgov/entity#26268](https://github.com/bcgov/entity/issues/26268). Thanks.

OK, so talked to Melissa and the requirements are to keep everything as is (hidden or shown button) with addition to currently disabling summaries for specified list

The ones that will be disabled in addition to being displayed are: `BC CC ULC C CCC CUL`

I moved back the v-if and added new disabled check.

Please let me know if you spot any additional issues.
Thanks !

severinbeauvais commented:
There's something wrong with this one. When its actions are disabled by the other FF, the Business Summary should not display at all.

https://business-dashboard-dev--pr-176-ar99hav8.web.app/C9900819
![image](https://github.com/user-attachments/assets/4607bc0e-a4a5-4b97-a7db-2c321235bfa4)

https://dev.business-dashboard.bcregistry.gov.bc.ca/C9900819:
![image](https://github.com/user-attachments/assets/bebc0aa9-f8c7-4d21-af78-b76141276cf7)

severinbeauvais commented:
I don't think you need a new FF. Does this logic make sense?

1. Whether to show the BS action: action is allowed (per API) + current business is not to be managed in COLIN (FF businesses-to-manage-in-colin).

2. Whether to disable the BS action: action is allowed (per API) + current business is a supported type (FF supported-business-summary-entities)

severinbeauvais reviewed (changes requested):
There is a logic error and a comment in 26268 that a new FF is needed, but don't think that's right.

severinbeauvais commented:
> 1. Whether to show the BS action = action is allowed (per API) + current business is not to be managed in COLIN (FF businesses-to-manage-in-colin).
> 2. Whether to disable the BS action = action is allowed (per API) + current business is a supported type (FF supported-business-summary-entities)

I'm still looking into this. I thought there was an allowable action for this but atm I don't see it.

hfekete commented:
> > 1. Whether to show the BS action = action is allowed (per API) + current business is not to be managed in COLIN (FF businesses-to-manage-in-colin).
> > 2. Whether to disable the BS action = action is allowed (per API) + current business is a supported type (FF supported-business-summary-entities)
> 
> ~I'm still looking into this. I thought there was an allowable action for this but atm I don't see it.~
> 
> There is no allowable action per the API, but there is a computed property which simply checks whether this is a business (ie, business id exists) (ie, it's not a bootstrap business with a Txxx number) and the `supported-business-summary-entities` FF.
> 
> So the new logic could be:
> 
> 1. Whether to show the BS action = this is a business + current business is not to be managed in COLIN (FF businesses-to-manage-in-colin).
> 2. Whether to disable the BS action = this is a business (redundant) + current business is a supported type (FF supported-business-summary-entities)

I am unsure on what types can appear form ([corp types](https://github.com/bcgov/bcrs-shared-components/blob/main/src/modules/corp-type-module/corp-type-module.ts))  in the dashboard.
if others can appear, then we need the 2nd for disabling. So we can verify corp type to show, and the other one to check which one is disabled.

At first I used only one FF but from your comments I understood that we might have different needs for different corp types to display/hide vs enable/disable.

The flag that is looking for corps managed in colin works against specific identifiers, rather than corp types, so I would not feel safe using it to exclude all the corp types we dont want to display.

currently the flag for supported summary holds: `CP BEN SP GP CBEN BC CC ULC C CCC CUL`
and the one for disabled ones: `BC CC ULC C CCC CUL`
Which has logic from the supported (1st ff), disable those in 2nd ff

Let me know if this makes sense.

severinbeauvais commented:
Thanks for your patience. I get it now: we don't want to show the BS action for _all_ businesses and disable for unsupported ones.

OK, let's go with a second FF. How about calling it `enabled-business-summary-entities` to make the logic easier to understand?

Is this now correct?

1. Whether to show the BS action = this is not a disabled corp (FF businesses-to-manage-in-colin) + this is a business + current business is a supported type (FF supported-business-summary-entities). (same as today)

2. Whether to disable the BS action = this is a business (redundant) + current business is enabled type (FF enabled-business-summary-entities).

hfekete commented:
/gcbrun

bcregistry-sre commented:
Temporary Url for review: https://business-dashboard-dev--pr-176-ar99hav8.web.app

hfekete commented:
> Temporary Url for review: https://business-dashboard-dev--pr-176-ar99hav8.web.app
> 
> BC LTD [BC0887777](https://business-dashboard-dev--pr-176-ar99hav8.web.app/BC0887777) BC ULC [BC0887778](https://business-dashboard-dev--pr-176-ar99hav8.web.app/BC0887778) BC CCC [BC0887779](https://business-dashboard-dev--pr-176-ar99hav8.web.app/BC0887779) Continuation in BC LTD [C9900907](https://business-dashboard-dev--pr-176-ar99hav8.web.app/C9900907) Continuation in BC ULC [C9900909](https://business-dashboard-dev--pr-176-ar99hav8.web.app/C9900909) Continuation in BC CCC [C9900908](https://business-dashboard-dev--pr-176-ar99hav8.web.app/C9900908)
> 
> has summary: GP [FM1060265](https://business-dashboard-dev--pr-176-ar99hav8.web.app/FM1060265)
> 
> has summary but excluded by another ff (not visible) [C9900819](https://business-dashboard-dev--pr-176-ar99hav8.web.app/C9900819)

If anyone has any additional TC let me know

hfekete commented:
> Thanks for your patience. I get it now: we don't want to show the BS action for _all_ businesses and disable for unsupported ones.
> 
> OK, let's go with a second FF. How about calling it `enabled-business-summary-entities` to make the logic easier to understand?
> 
> Is this now correct?
> 
> 1. Whether to show the BS action = this is not a disabled corp (FF businesses-to-manage-in-colin) + this is a business + current business is a supported type (FF supported-business-summary-entities). (same as today)
> 2. Whether to disable the BS action = this is a business (redundant) + current business is enabled type (FF enabled-business-summary-entities).

Appreciate all the comments, like we discussed offline I agree with your suggestion to name the flag enabled. Updated the code accordingly.

Let me know if anything else/new looks amiss.

severinbeauvais commented:
> If anyone has any additional TC let me know

Yup, these two. I just tested and they both display an enabled Business Summary action, as expected!

BEN - https://business-dashboard-dev--pr-176-ar99hav8.web.app/BC1230018
CBEN - https://business-dashboard-dev--pr-176-ar99hav8.web.app/C9900833

severinbeauvais reviewed (approved):
Looks great! 

~Don't merge yet as I'm trying to figure out whether to use a hotfix branch or whether we can take maain branch to Test and then Prod.~

_Updated: merged to main branch and then deployed to Test in preparation for current release!_
